### PR TITLE
Update Primevue to 4.0.5

### PR DESCRIPTION
**Pull request by wilwong89**
# Description
Update Primevue from 3.53.0 to 4.0.5.
Replace depreciated components
Update modified components
[PADS-234](https://apps.nrs.gov.bc.ca/int/jira/browse/PADS-234)
## Types of changes

New feature (non-breaking change which adds functionality)
Breaking change (fix or feature that would cause existing functionality to change) 

## Checklist

- [X] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [X] I have checked that unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments
Breaking change for Primevue v4. Will require running `npm ci`

### CODE REVIEW

**Comment by slhurley** [comment]
Lots of work here I see!  A couple of initial thoughts but I know the UX will give you the feedback you need....the menu bar at the top doesn't seem spaced correctly.  There is a big gap between Housing and Help in proponent view.  Add attachments on the Roadmap tab doesn't seem to work?  The top navigation through the tabs of the intake form aren't visually intuitive which tab you are on when you first load it (especially the first one since you don't have the lines to guide you).  The 3 buttons on the menu screen are all different in the centering.  I'll leave other comments to the UX eyes!

**Comment by kyle1morel** [comment]
Very quick side by side findings

1. Housing button on home page has no hover effect
2. Loading spinner is not centered
3. Housing buttons are centered aligned instead of left
4. Form card headers are not bold
5. All upload components have the previously hidden buttons instead of just a drag/drop area
6. No padding between Parcel ID/Geomark accordians
7. Submissions table has border below pagination. Can it be removed?
8. Table header color and create/search bar do not align with Dev

**Comment by Subin1Doo** [comment]
Things I noticed after a review of both proponent and nav side

Proponent view
General notes
- Padding uneven around Notice of collection message for both forms
- upload document function UI, can we revert it back to the old version?
- Tool tip message, background color is too light, makes it hard to read the text in white
- Placeholder texts, were they always that light in color? 

Intake form specific
- In the location tab - street address option - "search the address of your housing project" text is too small
- In the permits and reports tab - Provide additional location details section - PID and Geomark section expand arrow working the other way around. Arrow should point down when closed, point up when expanded. 
- In permits and reports tab - select all permits you think you may need section - padding around "add permit" button is uneven
- In permits and reports tab - padding uneven around "check this box if you agree to be contacted.."

Navigator view
- Files tab, upload document function UI, revert to old version
- Within a submission view, "Assigned to" field is small compared to other fields
- My bring forward notification expand/ close arrow, arrow should point down when closed, point up when expanded. 

I'll keep checking to see if I missed anything, thank you!

**Comment by wilwong89** [comment]
Posting Naomi's notes, she is having trouble logging into github.

Intake form
- Bug: Once on the Permits & Reports tab, the back button becomes “Back to my drafts and previous entries”, when it should be “Back to Housing”.
- Accessibility issues. Colour contrast need to be increased for the followings:
- The red alert message for missing fields – Can we change the text colour to a darker red? Maybe red-800 or 900?
- The green alert message for auto save – Can we change the text colour to a darker green, and bold the text?
- Tooltips – The grey background needs to be darker. Maybe use the darkest grey?
- The placeholder text need to be darker.
- Interaction issues. On the Location tab in the section for providing “Parcel ID” and “Geomark”, can we switch the direction of the arrow? The arrow should point down when collapsed, and up when expanded.

 
My Drafts and previous entries
- When hovering over a row in the list, the Confirmation ID is not highlighted. This is the same on Navs’ Submissions view. I guess this is the behaviour of primevue’s new design for the list/table component?
- When clicking on a row, it changes the text colour to blue. Can we disable this interaction? It doesn’t lead to anything other than changing the text colour.

 
Consistency
- Can we make the style of alert message boxes consistent? The red alert message for missing fields and the green alert message for autosave do not have the same style. Can we either remove the red line on the left from the red message box, or add a green line to the left of the green message box? If there’s a yellow alert message box, its style should be aligned as well.
- Can we make the hover colour consistent? Right now it uses grey for some components and light blue for others. There are also different hues of grey used for hovering among different components.

**Comment by qhanson55** [comment]
Found this in the SubmissionIntakeForm.vue file, not related to anything you did but could be deleted here maybe?
image

**Review by qhanson55 — APPROVED** [review]

**Comment by sanjaytkbabu** [comment]
image
1. Assigned to:  seems like its smaller than the others

File: frontend/src/components/housing/submission/SubmissionIntakeForm.vue
Comment (kyle1morel) [inline frontend/src/components/housing/submission/SubmissionIntakeForm.vue:544]: Assuming PrimeVue removed those useful slots so we dont have to manually track all this now?

File: frontend/src/components/housing/submission/SubmissionsNavigator.vue
Comment (kyle1morel) [inline frontend/src/components/housing/submission/SubmissionsNavigator.vue:387]: Remove commented code

File: frontend/src/components/housing/submission/SubmissionsNavigator.vue
Comment (kyle1morel) [inline frontend/src/components/housing/submission/SubmissionsNavigator.vue:280]: If `activeTabIndex` is no longer used, how are we defaulting to a tab based on the incoming query parameters?

File: frontend/src/components/layout/DevelopmentRoleOverride.vue
Comment (kyle1morel) [inline frontend/src/components/layout/DevelopmentRoleOverride.vue:49]: Replace `any` with correct type

File: frontend/src/views/DeveloperView.vue
Comment (kyle1morel) [inline frontend/src/views/DeveloperView.vue:31]: Use proper type over any

File: frontend/src/ThemeOverride.ts
Comment (wilwong89) [reply frontend/src/ThemeOverride.ts]: Removed

File: frontend/src/components/form/EditableSelect.vue
Comment (wilwong89) [reply frontend/src/components/form/EditableSelect.vue:78]: Fixed. Instead adjusted padding for the input to make sure it matches the height of the neighbouring select components.

File: frontend/src/components/form/InputNumber.vue
Comment (wilwong89) [reply frontend/src/components/form/InputNumber.vue:25]: Accidental over correction. Change undone.

File: frontend/src/components/form/internal/SelectInternal.vue
Comment (wilwong89) [reply frontend/src/components/form/internal/SelectInternal.vue:78]: Default size was insufficient, but 3rem is incorrect. Corrected to 1rem.

File: frontend/src/components/housing/submission/SubmissionIntakeForm.vue
Comment (wilwong89) [reply frontend/src/components/housing/submission/SubmissionIntakeForm.vue:544]: Yes, the top level `<Stepper :value="activeStep">` component tracks the step through the `value` prop.

File: frontend/src/components/housing/submission/SubmissionsNavigator.vue
Comment (wilwong89) [reply frontend/src/components/housing/submission/SubmissionsNavigator.vue:280]: Fixed by setting top level Tabs component to watch for activeTabIndex again and setting the various tabs to numbers so it can respond to query parameters.

File: frontend/src/components/layout/DevelopmentRoleOverride.vue
Comment (wilwong89) [reply frontend/src/components/layout/DevelopmentRoleOverride.vue:49]: Added type

File: frontend/src/views/DeveloperView.vue
Comment (wilwong89) [reply frontend/src/views/DeveloperView.vue:31]: Added type

File: frontend/src/components/housing/submission/SubmissionIntakeForm.vue
Comment (wilwong89) [reply frontend/src/components/housing/submission/SubmissionIntakeForm.vue:597]: Changed so that a watcher tracks step changes and triggers `onSaveDraft` accordingly.

File: frontend/src/assets/main.scss
Comment (kyle1morel) [inline frontend/src/assets/main.scss:147]: Can these not be set in the theme override?

File: frontend/src/assets/variables.scss
Comment (kyle1morel) [inline frontend/src/assets/variables.scss:6]: If this is unused it can be removed.

File: frontend/src/assets/variables.scss
Comment (kyle1morel) [inline frontend/src/assets/variables.scss:19]: I believe some of these can be removed as they can either be accessible via `var(--p-primary-color)`, are just unused, or maybe exist once and can probably be replaced via the first method. Will leave it to you to evaluate.

File: frontend/src/components/form/DatePicker.vue
Comment (kyle1morel) [inline frontend/src/components/form/DatePicker.vue:27]: Is PrimeVue now handling the date internally as a string?

File: frontend/src/components/form/StepperNavigation.vue
Comment (kyle1morel) [inline frontend/src/components/form/StepperNavigation.vue:20]: TODO

File: frontend/src/components/form/internal/SelectInternal.vue
Comment (kyle1morel) [reply frontend/src/components/form/internal/SelectInternal.vue:78]: Still seeing 3rem

File: frontend/src/components/layout/Navbar.vue
Comment (kyle1morel) [inline frontend/src/components/layout/Navbar.vue:181]: Unused code can be removed.

File: frontend/src/lib/primevue/index.ts
Comment (kyle1morel) [inline frontend/src/lib/primevue/index.ts:55]: Remove

File: frontend/src/views/HomeView.vue
Comment (kyle1morel) [inline frontend/src/views/HomeView.vue:55]: Ref comment in variable.scss about updating these to use `var()`

File: frontend/src/lib/primevue/index.ts
Comment (wilwong89) [reply frontend/src/lib/primevue/index.ts:55]: Removed extra code

File: frontend/src/components/layout/Navbar.vue
Comment (wilwong89) [reply frontend/src/components/layout/Navbar.vue:181]: Removed extra code

File: frontend/src/components/form/StepperNavigation.vue
Comment (wilwong89) [reply frontend/src/components/form/StepperNavigation.vue:20]: Removed extra code, was done.

File: frontend/src/components/form/DatePicker.vue
Comment (wilwong89) [reply frontend/src/components/form/DatePicker.vue:27]: Fixed accidental replace, Date type is back.

File: frontend/src/assets/variables.scss
Comment (wilwong89) [reply frontend/src/assets/variables.scss:19]: Moved all variables in this file to `ThemeOverride.ts` file. Fixed all references to new system. Removed extraneous `variables.scss` file.

File: frontend/src/assets/variables.scss
Comment (wilwong89) [reply frontend/src/assets/variables.scss:6]: Removed extra code

File: frontend/src/views/HomeView.vue
Comment (wilwong89) [reply frontend/src/views/HomeView.vue:55]: Changed to utilize `var()` in call cases.

**Comment by wilwong89** [comment]
Temporarily putting this ticket on hold. Will revisit at a later date.

[PADS-234](https://apps.nrs.gov.bc.ca/int/jira/browse/PADS-234)

**Pull request closed without merging**
